bare_metal_tests/TESTS/bringup/uart/test.py

```diff
@@ -0,0 +1,330 @@
+            print('json ok')
+    except:
+        logging.error('Unable to locate image file.')
+        exit()
+       
+    return image_path
+    
+def reset_mbed_board(port, delay_reset):
+    """Perform reset of the mbed device by sending
+       a break signal to the device.
+       
+    @param port serial port id.
+    @param delay_reset delay time after reset [s].
+    """
+    logging.info('Reset the mbed board...')
+ 
+    break_port = serial.Serial(port=port, baudrate=9600)
+    
+    break_port.sendBreak()
+
+    break_port.close()
+    
+    logging.info('Wait %f sec after the reset.' % (delay_reset))
+    time.sleep(delay_reset)
+    
+def flash_mbed_board(src, dst, delay):
+    """Perform flash of the mbed device.
+    
+    @param src path to the .bin file.
+    @param dst path to mbed device.
+    @param delay delay time after flashing [s].
+    """
+    logging.info('Flashing mbed board...')
+    logging.info('source: %s' % (src))
+    logging.info('destination: %s' % (dst))
+    
+    shutil.copyfile(src, dst)
+    
+    logging.info('Wait %f sec after flashing.' % (delay))
+    time.sleep(delay)
+    
+def build_test(target, toolchain, clean_flag):
+    """Build the mbed test.
+    
+    @param target target id.
+    @param toolchain compile toolchain.
+    @return image path.
+    """
+    logging.info('Building mbed UART test for green tea.')
+    
+    clean = ''
+    if(clean_flag):
+        clean = '-c'
+    
+    cmd = 'mbed test -t %s -m %s --source . --source bare_metal_tests --build-data bare_metal_tests/TESTS/bringup/uart/build.log --compile %s' % (toolchain, target, clean)
+    return run_command(cmd)
+
+def open_port(port_id, baudrate):
+    """Open serial port.
+    
+    @param port_id serial port id.
+    @param baudrate serial port baudrate.
+    """
+    global serial_port
+    try:
+        serial_port = serial.Serial(
+        port=port_id,
+        baudrate=baudrate,
+        parity=serial.PARITY_NONE,
+        stopbits=serial.STOPBITS_ONE,
+        bytesize=serial.EIGHTBITS)
+
+    except:
+        logging.error('Could not open serial port.')
+        exit()
+
+def serial_write_line(line):
+    """Write line to the serial port.
+    
+    @param line line to be written.
+    """
+    serial_port.write(line)
+    logging.debug('[SEDN] %s' % line)
+
+def serial_read_line():
+    """Read line from the serial port with timeout.
+    
+    @return string read from the serial port.
+    """
+    global serial_port
+    limit = 0
+    line = str()
+    while(True):
+        if(serial_port.in_waiting == 0):
+            time.sleep(0.001)
+            limit += 1
+            if (limit == RETRY_LIMIT):
+                logging.debug('[RECV] %s' % line)
+                return line
+            continue
+        limit = 0
+        c = serial_port.read()
+        line += c
+        if(c == '\0'):
+            logging.debug('[RECV] %s' % line)
+            return line
```

Comment:
This line, and the prior return look to be the same. Could you de-duplicate this code? Further, the `if ... continue` is the same as an `if ... else`. I think that an `if  ... else` would be more clear here.
For example:
```python
    while(True):
        if(serial_port.in_waiting == 0):
            time.sleep(0.001)
            limit += 1
            if (limit == RETRY_LIMIT):
                break
        else:
            limit = 0
            c = serial_port.read()
            line += c
            if(c == '\0'):
                break
    logging.debug('[RECV] %s' % line)
    return line

```

Reply:
Fixed as suggested.